zipline/_protocol.pyx

```diff
@@ -76,6 +79,289 @@ cdef class BarData:
+        the current trade bar.  If there is no current trade bar, NaN is
+        returned.
+        """
+        if isinstance(assets, Asset):
+            if isinstance(fields, str):
+                # return scalar value
+                return self.data_portal.get_spot_value(
+                    assets,
+                    fields,
+                    self.simulation_dt_func(),
+                    self.data_frequency
+                )
+            else:
+                # assume fields is iterable
+                # return a Series indexed by field
+                return pd.Series(data={
+                    field: self.data_portal.get_spot_value(
+                        assets,
+                        field,
+                        self.simulation_dt_func(),
+                        self.data_frequency)
+                    for field in fields
+                }, index=fields, name=assets.symbol)
+        else:
+            if isinstance(fields, str):
+                # assume assets is iterable
+                # return a Series indexed by asset
+                return pd.Series(data={
+                    asset: self.data_portal.get_spot_value(
+                        asset,
+                        fields,
+                        self.simulation_dt_func(),
+                        self.data_frequency)
+                    for asset in assets
+                    }, index=assets, name=fields)
+            else:
+                # both assets and fields are iterable
+                data = {}
+
+                for field in fields:
+                    series = pd.Series(data={
+                        asset: self.data_portal.get_spot_value(
+                            asset,
+                            field,
+                            self.simulation_dt_func(),
+                            self.data_frequency)
+                        for asset in assets
+                        }, index=assets, name=field)
+
+                    data[field] = series
+
+                return pd.DataFrame(data)
+
+    def can_trade(self, assets):
+        """
+        For the given asset or iterable of assets, returns true if the asset
+        is alive at the current simulation time and there is a known last
+        price.
+
+        Parameters
+        ----------
+        assets: Asset or iterable of assets
+
+        Returns
+        -------
+        boolean or Series of booleans, indexed by asset.
+        """
+        dt = self.simulation_dt_func()
+        data_portal = self.data_portal
+
+        if isinstance(assets, Asset):
+            return self._can_trade_for_asset(assets, dt, data_portal)
+        else:
+            return pd.Series(data={
+                asset: self._can_trade_for_asset(asset, dt, data_portal)
+                for asset in assets
+            })
+
+    cdef _can_trade_for_asset(self, asset, dt, data_portal):
+        if asset.start_date > dt:
```

Comment:
done, but your logic there isn't right (we need to check if there is a previous price, not if volume != 0)